### ISSUE TITLE
Add Visual Studio Code workspace to project root

### ISSUE THREAD

**Issue by radium-v** [microsoft/fast#3136]
**What feature are you requesting?**
It would be great if we could add a [`.vscode` workspace directory](https://code.visualstudio.com/docs/getstarted/settings) to the project root.

**What would be the benefit of adding this feature?**
It seems that our team primarily uses VSC, and this would allow us to recommend extensions for syntax highlighting and also set some editor configurations 

**What alternatives have you considered?**
There are some more environment-agnostic options like `editorconfig`, which could work as an alternative option for users who don't use VSC.

**Is there any additional context for your request?**
Certain VSC settings can conflict or interfere with the tools we use - for instance, Prettier doesn't catch things in all files, and things like indentation or EOL newline can be enforced to reduce the dependence on Prettier.

**Comment by radium-v** [comment]
Here are some defaults I'm thinking about:

```json
{
  "[typescript]": {
    "editor.tabSize": 4
  },
  "editor.trimAutoWhitespace": true,
  "files.eol": "\n",
  "files.trimTrailingWhitespace": true,
  "files.insertFinalNewline": true,
  "files.trimFinalNewlines": true,
  "javascript.format.enable": false,
  "javascript.format.insertSpaceAfterFunctionKeywordForAnonymousFunctions": false,
  "js/ts.implicitProjectConfig.experimentalDecorators": true,
  "npm.packageManager": "yarn",
  "typescript.enablePromptUseWorkspaceTsdk": true,
  "typescript.format.enable": false,
  "typescript.preferences.quoteStyle": "double"
}
```

**Comment by awentzel** [comment]
I'm in favor of adding this as I've been using for a while myself. We can add "cSpell.words" and "cSpell.ignoreWords" to eliminate common workspace spelling nuances.  I also will require this potentially in the future for different Azure packages that can be managed within VSCode.

We should add a task to meet with the team and see what we can determine are safe defaults we'd want to implement across packages.

**Comment by radium-v** [comment]
This is resolved via #4574. @awentzel, can you create a new issue for adding spellcheck?

**Comment by awentzel** [comment]
Per note above by Radium-v. Resolving this PR.
